lisa/nic.py

```diff
@@ -456,3 +483,81 @@ def _get_default_nic(self) -> None:
         ).is_true()
         self.default_nic: str = default_interface_name
         self.default_nic_route = dev_match.group()
+
+    def is_mana_device_discovered(self) -> bool:
+        pci_devices = self._node.tools[Lspci].get_devices_by_type(
+            constants.DEVICE_TYPE_SRIOV
+        )
+        if (
+            pci_devices
+            and "Device 00ba" in pci_devices[0].device_info
+            and pci_devices[0].vendor == "Microsoft Corporation"
+        ):
+            return True
+        return False
+
+    def is_mlx_device(self) -> bool:
+        pci_devices = self._node.tools[Lspci].get_devices_by_type(
+            constants.DEVICE_TYPE_SRIOV
+        )
+        if pci_devices and "Mellanox Technologies" in pci_devices[0].device_info:
+            return True
+        return False
+
+    def is_mana_enabled(self) -> bool:
+        if self.is_mana_device_discovered() and self._node.tools[
+            KernelConfig
+        ].is_enabled("CONFIG_MICROSOFT_MANA"):
+            return True
+        return False
+
+    def exists_vf(self) -> bool:
+        if self.is_mlx_device() or self.is_mana_enabled():
+            return True
+        return False
+
+    def expected_vf_count(self) -> int:
```

Comment:
We should be clear on different concepts. 
1. Here looks like should be "secondary", but I don't know why if no vf exists, it should return 1.
2. Why the lower nics represent the "vf"? When mana is functional inside of synthetic. Should it be a VF as well?

Reply:
for mlx, if there are 6 sriov nics, then lspci will return 6 pci devices. so previously we count expected value by count length of total sriov nic (paired nic of synthetic nic)
for mana, it is always 1.